src/main/java/com/arcaneminecraft/bungee/JoinLeaveEvents.java

```diff
@@ -0,0 +1,165 @@
+import net.md_5.bungee.api.chat.TranslatableComponent;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+import net.md_5.bungee.api.event.PlayerDisconnectEvent;
+import net.md_5.bungee.api.event.PostLoginEvent;
+import net.md_5.bungee.api.plugin.Listener;
+import net.md_5.bungee.event.EventHandler;
+
+import java.text.DateFormat;
+import java.util.TimeZone;
+import java.util.concurrent.TimeUnit;
+
+public class JoinLeaveEvents implements Listener {
+    private final ArcaneBungee plugin;
+    private final BaseComponent welcomeMessage;
+
+
+    JoinLeaveEvents(ArcaneBungee plugin) {
+        this.plugin = plugin;
+
+        // BEGIN Welcome Message construction
+        BaseComponent temp;
+        TextComponent line = new TextComponent(new String(new char[27]).replace("\0", " "));
+        line.setColor(ChatColor.GRAY);
+        line.setStrikethrough(true);
+        this.welcomeMessage = new TextComponent("\n");
+
+        // First line
+        this.welcomeMessage.addExtra(line);
+        temp = new TextComponent(" Arcane Survival ");
+        temp.setColor(ArcaneColor.HEADING);
+        temp.setBold(true);
+        this.welcomeMessage.addExtra(temp);
+
+        this.welcomeMessage.addExtra(line);
+        this.welcomeMessage.addExtra("\n\n");
+
+
+        // help line
+        this.welcomeMessage.addExtra("            You can type ");
+        temp = new TextComponent("/help");
+        temp.setColor(ArcaneColor.HEADING);
+        temp.setBold(true);
+        temp.setClickEvent(new ClickEvent(ClickEvent.Action.SUGGEST_COMMAND,"/help"));
+        this.welcomeMessage.addExtra(temp);
+        this.welcomeMessage.addExtra(" for a list of commands.\n");
+
+        // Website
+        this.welcomeMessage.addExtra("        Visit our website at ");
+        temp = ArcaneText.urlSingle("https://arcaneminecraft.com/");
+        temp.setColor(ArcaneColor.HEADING);
+        this.welcomeMessage.addExtra(temp);
+        this.welcomeMessage.addExtra("!\n\n");
+
+        temp = new TextComponent(new String(new char[80]).replace("\0", " "));
+        temp.copyFormatting(line);
+        this.welcomeMessage.addExtra(temp);
+        this.welcomeMessage.addExtra("\n");
+    }
+
+    @EventHandler
+    public void onLoginJoin(PostLoginEvent e) {
+        if (plugin.getSqlDatabase() != null) {
+            plugin.getSqlDatabase().playerJoinThen(e.getPlayer(), (time, oldName) -> {
+                ProxiedPlayer p = e.getPlayer();
+                if (OptionsStorage.get(p, OptionsStorage.Toggles.SHOW_WELCOME_MESSAGE)) {
+                    p.sendMessage(ChatMessageType.SYSTEM, this.welcomeMessage);
+                }
+
+                if (p.hasPermission("arcane.welcome.donor") && OptionsStorage.get(p, OptionsStorage.Toggles.SHOW_DONOR_WELCOME_MESSAGE)) { // TODO: Check permission node existence on ArcaneServer
+                    p.sendMessage(ChatMessageType.SYSTEM, new TextComponent("You are a donor! This message is still a work in progress"));
```

Comment:
"This message is still a work in progress" ? Seems fine to me. Maybe mention a donor help command if we've got that as a separate thing?

Reply:
As we discussed in chat, I forgot to port [this](https://github.com/ArcaneMinecraft/ArcaneDonor/blob/master/src/main/java/com/arcaneminecraft/donor/ArcaneDonor.java#L94) over :I